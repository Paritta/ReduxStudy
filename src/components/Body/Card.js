import React from "react";
import styled from "styled-components";
import EmptyImage from "../../asset/EmptyImage.png";
import PropTypes from "prop-types";
import ReactLoading from "react-loading";

const propTypes = {
    image: PropTypes.object,
    imageReceiveRequest: PropTypes.func,
    ImageReceive: PropTypes.object,
};

const defaultTypes = {
    image: {},
    imageReceiveRequest() {},
    ImageReceive: {},
};

const Wrapper = styled.div`
    margin: 0;
    padding: 0;
    
    width: 450px;
    border-radius: 10px 10px 0 0;
    overflow: hidden;
`;

const View = styled.div`
    margin: 0;
    padding: 0;

    width: 450px;
    height: 250px;
    
    background: url(${props => props.ImageUrl ? props.ImageUrl : EmptyImage}) center center;
    background-size: ${props => props.ImageUrl ? 100 : 40}px;
    
    display: flex;
    align-items: center;
    justify-content: center;
    
    color: white;
    
    transform: scale(10);
    
    z-index: 5;
`;

const Info =styled.div`
    margin: 0;
    padding: 0;
    
    overflow: hidden;
    width: 450px;
    height: 100px;
    
    background: hsla(0, 72%, 0%, 0.4);
    
    z-index: 10;
    
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
`;

const Title = styled.div`
    font-family: 'Hanna', fantasy;
    font-size: 40px;
    font-weight: 500;
    color: white;
`;

const WrappedReactLoading = styled.div`
    transform: scale(2);
    position: absolute;
    
    top: 15%;
    left: 40%;
    
    padding: 80px 0;
`;

const LocationDiv = styled.div`   
    font-family: 'Hanna', fantasy;
    font-size: 22px;
    font-weight: 400;
    color: white;
`;

export class Card extends React.Component {
    componentDidMount () {
        this.props.imageReceiveRequest(this.props.postId)
    }

    render () {
        const { item, ImageReceive } = this.props;
        const { StudyTitle, Location } = item.data.values;
        const Pending = ImageReceive.pending;
        const ImageData = ImageReceive.data;
        let ImageUrl = false;
        if(this.props.key !== undefined && ImageReceive.data.length !== 0 && ImageReceive.data !== undefined) {
            ImageUrl = ImageReceive.data[this.props.key].Url;
        }
        const PostImageKey = item.data.PostImageKey;

        return (
            <div>
                {/* 이미지 로딩 */}
                {
                    Pending && ImageData.length === 0 &&
                    <WrappedReactLoading>
                        <ReactLoading type="cylon" color="palevioletred"/>
                    </WrappedReactLoading>
                }

                {/* 이미지가 없을 때는 EmptyImage  */}
                {
                    PostImageKey === undefined &&
                    <Wrapper>
                        <View
                            ImageUrl={false}
                        >
                            .
                        </View>
                    </Wrapper>
                }

                {/* 이미지가 있을 때는 ImageUrl, 이미지 등록 버튼 삭제 */}
                {
                    PostImageKey !== undefined &&
                    <Wrapper>
                        <View
                            ImageUrl={ImageUrl}
                        >
                        </View>
                    </Wrapper>
                }
                {/*<Wrapper>*/}
                    {/*<View>*/}
                        {/*<FaCameraRetro/>*/}
                    {/*</View>*/}
                {/*</Wrapper>*/}

                <Info>
                    <Title>
                        { StudyTitle }
                    </Title>
                    <LocationDiv>
                        { Location }
                    </LocationDiv>
                </Info>
            </div>
        );
    }
}

Card.propTypes = propTypes;
Card.defaultTypes = defaultTypes;

export default Card;