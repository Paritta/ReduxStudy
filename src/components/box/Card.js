import React from "react";
import styled from "styled-components";
import oc from "open-color";
import FaCameraRetro from "react-icons/lib/fa/camera-retro";

const propTypes = {
};

const defaultTypes = {
};

const Wrapper = styled.div`
    display: inline-block;
    
    margin: 0;
    padding: 0;
    
    border-radius: 15px;
    overflow: hidden;
`;

const View = styled.div`
    margin: 0;
    padding: 0;

    width: 450px;
    height: 300px;
    
    background: ${oc.gray[1]};
    
    display: flex;
    align-items: center;
    justify-content: center;
    
    color: white;
    
    transform: scale(10);
    
    z-index: 5;
`;

const Info =styled.div`
    margin: 0;
    padding: 0;
    
    overflow: hidden;
    width: 450px;
    height: 100px;
    
    background: pink;
    
    z-index: 10;
`;

const Card = () => {
    return (
        <div>
            <Wrapper>
                <View>
                    <FaCameraRetro/>
                </View>
            </Wrapper>
            <Info/>
        </div>
    );
};

Card.propTypes = propTypes;
Card.defaultTypes = defaultTypes;

export default Card;