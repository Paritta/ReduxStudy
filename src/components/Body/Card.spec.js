import React from "react";
import { Card } from "./Card";
import "jest-styled-components";
import ReactTestRenderer from "react-test-renderer";

it("Card render correctly", () => {
    const item = {
        data: {
            values: {
            },
            Image: {
                Url: {
                    
                }
            }
        }
    };

    const ImageReceive = {
        pending: {

        },
        data : {
            length: 0
        }
    };

    function imageReceiveRequest () {

    }

    const tree = ReactTestRenderer.create(
        <Card
            item={item}
            ImageReceive={ImageReceive}
            imageReceiveRequest={imageReceiveRequest}
        />
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
