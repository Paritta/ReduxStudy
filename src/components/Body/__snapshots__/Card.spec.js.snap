// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render correctly 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  width: 450px;
  border-radius: 10px 10px 0 0;
  overflow: hidden;
}

.c1 {
  margin: 0;
  padding: 0;
  width: 450px;
  height: 300px;
  background: #f1f3f5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: white;
  -webkit-transform: scale(10);
  -ms-transform: scale(10);
  transform: scale(10);
  z-index: 5;
}

.c2 {
  margin: 0;
  padding: 0;
  overflow: hidden;
  width: 450px;
  height: 100px;
  background: hsla(0,72%,0%,0.4);
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  font-size: 40px;
  font-weight: 500;
  color: white;
}

.c4 {
  font-size: 22px;
  font-weight: 400;
  color: white;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m20.7 18.6q0-0.3-0.2-0.5t-0.5-0.2q-1.5 0-2.5 1t-1.1 2.5q0 0.3 0.2 0.5t0.5 0.2 0.6-0.2 0.2-0.5q0-0.9 0.6-1.5t1.5-0.6q0.3 0 0.5-0.2t0.2-0.5z m5 2.9q0 2.3-1.7 4t-4 1.7-4-1.7-1.7-4 1.7-4.1 4-1.6 4 1.6 1.7 4.1z m-22.8 12.8h34.2v-2.9h-34.2v2.9z m25.7-12.8q0-3.6-2.5-6.1t-6.1-2.5-6.1 2.5-2.5 6.1 2.5 6 6.1 2.5 6.1-2.5 2.5-6z m-22.9-14.4h8.6v-2.8h-8.6v2.8z m-2.8 4.3h34.2v-5.7h-18.4l-1.5 2.9h-14.3v2.8z m37.1-5.7v28.6q0 1.2-0.8 2t-2.1 0.8h-34.2q-1.2 0-2.1-0.8t-0.8-2v-28.6q0-1.2 0.8-2t2.1-0.8h34.2q1.2 0 2.1 0.8t0.8 2z"
          />
        </g>
      </svg>
    </div>
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
    />
    <div
      className="c4"
    />
  </div>
</div>
`;
